sys/include/sema.h

```diff
@@ -33,21 +35,37 @@ extern "C" {
 /**
  * @brief   Creates semaphore statically.
  *
- * @param[in] value Initial value for the semaphore.
+ * The value for the statically created semaphore must be 0
  *
  * @return  Statically initialized semaphore.
  */
-#define SEMA_CREATE(value)      { (value), PRIORITY_QUEUE_INIT }
+#define SEMA_CREATE()           { (0), SEMA_OK, MUTEX_INIT() }
+
+/**
+ * @brief A Semaphore states.
+ */
+typedef enum {
+    SEMA_OK = 0,
+    SEMA_DESTROY,
+} sema_state_t;
 
 /**
  * @brief A Semaphore.
  */
 typedef struct {
-    volatile unsigned int value;    /**< value of the semaphore */
-    priority_queue_t queue;         /**< list of threads waiting for the semaphore */
+    atomic_int_t value;             /**< value of the semaphore */
+    sema_state_t state;             /**< state of the semaphore */
+    mutex_t mutex;                  /**< mutex of the semaphore */
```

Comment:
mh.... actually I meant it the other way around. `sema` used to be based on flags (or rather `posix_semaphore` was before we took the base implementation out of the POSIX wrapper and made it message-based) and it didn't work out as we thought it might (see also size of the current `sema_t` struct vs how it was before).